src/providers/twitch/TwitchAccount.cpp

```diff
@@ -354,42 +358,88 @@ SharedAccessGuard<const TwitchAccount::TwitchAccountEmoteData>
 }
 
 // AutoModActions
-void TwitchAccount::autoModAllow(const QString msgID)
+void TwitchAccount::autoModAllow(const QString msgId, ChannelPtr channel)
 {
-    QString url("https://api.twitch.tv/kraken/chat/twitchbot/approve");
+    getHelix()->manageAutoModMessages(
+        msgId, QString("ALLOW"),
+        [] {
+            // success
+        },
+        [channel](auto error) {
+            // failure
+            QString errorMessage("Failed to allow AutoMod message - ");
 
-    auto qba = (QString("{\"msg_id\":\"") + msgID + "\"}").toUtf8();
+            switch (error)
+            {
+                case HelixAutoModMessageError::MessageAlreadyProcessed: {
+                    errorMessage += "message has already been processed.";
+                }
+                break;
 
-    NetworkRequest(url, NetworkRequestType::Post)
-        .header("Content-Type", "application/json")
-        .header("Content-Length", QByteArray::number(qba.size()))
-        .payload(qba)
+                case HelixAutoModMessageError::UserNotAuthorized: {
+                    errorMessage +=
+                        "you don't have permission to perform that action";
+                }
+                break;
 
-        .authorizeTwitchV5(this->getOAuthClient(), this->getOAuthToken())
-        .onError([=](NetworkResult result) {
-            qCWarning(chatterinoTwitch)
-                << "[TwitchAccounts::autoModAllow] Error" << result.status();
-        })
-        .execute();
+                case HelixAutoModMessageError::MessageNotFound: {
+                    errorMessage += "target message not found.";
+                }
+                break;
+
+                // This would most likely happen if the service is down, or if the JSON payload returned has changed format
+                case HelixAutoModMessageError::Unknown:
+                default: {
+                    errorMessage += "an unknown error occured. You may need to "
```

Comment:
Error message doesn't match the description in the comment above. Maybe a better message is `"an unknown error occurred. The response from Twitch was not as expected"`

The user re-authing would fix this the majority of the time (unless I'm missing something because of that comment)

Reply:
We just return a generic error response, same as in /marker command in CommandController.cpp the comment is a copy-paste from there. No idea if we have to change this in any way or not.

I also added proper handling for 401 now, so it's just an unknown error.